### DISCUSSION TITLE
Blockers for implementation. Future of project

### DISCUSSION THREAD

**Discussion by kinseii** [shalb/cluster.dev#286] [General]
I would like to discuss the possible future of the project, how actively it is developing, what plans the creators have. I am currently creating a PoC for my company based on Cluster.dev and have encountered some problems and have some questions. I have created several Issue, I got an answer to the first ones, but I don't get an answer to the last ones. I really want to make our infrastructure on Cluster.dev, however there are some blockers:

- I cannot pass to the input of a kubernetes unit (with `kind: Secret`) the output from another module (Issue: https://github.com/shalb/cluster.dev/issues/285).
- I can't pass a string to the input of the tfmodule unit via the `readFile "file"` function, so that I can use variable definitions from remoteState within the file. I think it can be done through string templating (Issue: https://github.com/shalb/cluster.dev/issues/225).
- And there is also a very big problem with how to deploy different parts of the infrastructure if there are several people working on the project. The thing is that using terragrunt we can deploy modules selectively, using the options `--terragrunt-include-dir`, `-terragrunt-exclude-dir`, `-terragrunt-strict-include`, etc. Or even by going into the desired module and deploying just that one. Cluster.dev doesn't allow you to do this yet, and as a result there will be problems over the fact that while one engineer is working on one StackTemplate or even a unit, another engineer will have to wait so as not to roll back his changes. In this case, I suggest implementing options specifying specific Stack, StackTemplate, and even Units.

**Comment by voatsap** [comment]
Hi @kinseii we have some vacations, so there was some slowness with replies, now we are back on track, will go through and reply on your issues.

**Reply by kinseii** [reply]
Hi! Regarding the 3rd feature, I created request: https://github.com/shalb/cluster.dev/issues/288
